multi_partner_learning.py

```diff
@@ -448,29 +494,48 @@ def build_model_from_weights(self, new_weights):
         )
 
         return new_model
-
-    def init_with_new_models(self):
+    
+    
+    def init_with_models(self, use_weights_from_previous_coalition=False):
         """Return a list of newly generated models, one per partner"""
 
         # Init a list to receive a new model for each partner
         partners_model_list = []
 
         # Generate a new model and add it to the list
-        new_model = self.generate_new_model()
-        partners_model_list.append(new_model)
+        if use_weights_from_previous_coalition:
+            new_model = self.generate_new_model()
+            new_model.load_weights(self.init_model_from)
+            model_weights = new_model.get_weights()
+            partners_model_list.append(new_model)
```

Comment:
Here it seems that the model is added to the list before being compiled with its new weights. Or does `load_weights()` replaces `set_weights() + .compile()` ?